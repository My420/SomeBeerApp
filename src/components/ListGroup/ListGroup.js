import React from 'react';
import PropTypes from 'prop-types';
import './listgroup.scss';

const ListGroup = ({ children, className, tag: Tag, ...attr }) => {
  return (
    <Tag className={className} {...attr}>
      {children}
    </Tag>
  );
};

export default ListGroup;

ListGroup.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string])
};

ListGroup.defaultProps = {
  children: null,
  className: 'list-group',
  tag: 'ul'
};
