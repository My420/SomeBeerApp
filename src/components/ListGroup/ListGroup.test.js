import React from 'react';
import { shallow } from 'enzyme';
import ListGroup from './ListGroup';

describe('Test <ListGroup /> component', () => {
  describe('Test <ListGroup /> without props', () => {
    const wrapper = shallow(<ListGroup />);

    test('should correct render component without props', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should  render tag <ul> by default', () => {
      expect(wrapper.find('ul')).toHaveLength(1);
    });
  });

  describe('Test <ListGroup /> with props', () => {
    const wrapper = shallow(
      <ListGroup className="test" tag="ol" data-test="test">
        <li>test</li>
        <li>test</li>
      </ListGroup>
    );

    test('should correct render component with props', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should set correct tag with correct className', () => {
      expect(wrapper.find('ol').filter('.test')).toHaveLength(1);
    });

    test('should transfer attribute', () => {
      expect(wrapper.find('ol').prop('data-test')).toBe('test');
    });

    test('should render children', () => {
      expect(wrapper.find('li')).toHaveLength(2);
    });
  });
});
