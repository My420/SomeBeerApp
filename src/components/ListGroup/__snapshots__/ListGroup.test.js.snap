// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ListGroup /> component Test <ListGroup /> with props should correct render component with props 1`] = `
<ol
  className="test"
  data-test="test"
>
  <li>
    test
  </li>
  <li>
    test
  </li>
</ol>
`;

exports[`Test <ListGroup /> component Test <ListGroup /> without props should correct render component without props 1`] = `
<ul
  className="list-group"
/>
`;
